### PR TITLE
[test] Do not override downloaded envoy if already exists.

### PR DESCRIPTION
**Pull request by bianpengyuan**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: test/envoye2e/driver/envoy.go [hunk 1]
```diff
@@ -233,7 +233,7 @@ func downloadEnvoy(ver string) (string, error) {
 	}
 	defer os.RemoveAll("usr/")
 
-	cpCmd := exec.Command("cp", src, dst)
+	cpCmd := exec.Command("cp", "-n", src, dst)
```
Comment (PiotrSikora) [inline test/envoye2e/driver/envoy.go:236]: Shouldn't this check be performed before we download Envoy? It's a bit wasteful to download all those bytes if we're not going to use the binary.

**Review by PiotrSikora — CHANGES_REQUESTED** [review]

File: test/envoye2e/driver/envoy.go (hunk 1 above)
Comment (bianpengyuan) [reply test/envoye2e/driver/envoy.go:236]: There is already a check at line 209, but duplicate download still happens some time due to test runs parallel.

Comment (PiotrSikora) [reply test/envoye2e/driver/envoy.go:236]: Ah, OK.

**Review by PiotrSikora — APPROVED** [review]
.

**Pull request merged**